src/StorageFlattening.cpp

```diff
@@ -339,77 +340,80 @@ class FlattenDimensions : public IRMutator {
             }
         }
 
-        stmt = Evaluate::make(Call::make(op->types[0], Call::prefetch, args, Call::Intrinsic));
+        return Evaluate::make(Call::make(op->types[0], Call::prefetch, args, Call::Intrinsic));
     }
 
-    void visit(const LetStmt *let) {
+    Stmt visit(const LetStmt *op) override {
         // Discover constrained versions of things.
-        bool constrained_version_exists = ends_with(let->name, ".constrained");
+        bool constrained_version_exists = ends_with(op->name, ".constrained");
         if (constrained_version_exists) {
-            scope.push(let->name, 0);
```

Comment:
Not for this PR, but conditionally pushing something occurs a lot, and it's an awkward idiom because you need to track whether or not to pop it. It would be nice if ScopedBinding were nullable -i.e. had a default constructor so that you could declare one at function scope, and then conditionally assign to it inside an if clause.

Reply:
Agreed.